lib/services/android-plugin-build-service.ts

```diff
@@ -0,0 +1,359 @@
+		const manifest = path.join(platformsDir, AndroidPluginBuildService.ANDROID_MANIFEST_XML);
+		return this.$fs.exists(manifest) ? manifest : null;
+	}
+
+	private getShortPluginName(pluginName: string): string {
+		return pluginName.replace(/[\-]/g, "_");
+	}
+
+	private async updateManifestContent(oldManifestContent: string, defaultPackageName: string): Promise<string> {
+		const content = oldManifestContent;
+		let newManifestContent;
+
+		let xml: any = await this.getXml(content);
+
+		let packageName = defaultPackageName;
+		// if the manifest file is full-featured and declares settings inside the manifest scope
+		if (xml["manifest"]) {
+			if (xml["manifest"]["$"]["package"]) {
+				packageName = xml["manifest"]["$"]["package"];
+			}
+
+			// set the xml as the value to iterate over its properties
+			xml = xml["manifest"];
+		}
+
+		// if the manifest file doesn't have a <manifest> scope, only the first setting will be picked up
+		const newManifest: any = { manifest: {} };
+		for (const prop in xml) {
+			newManifest.manifest[prop] = xml[prop];
+		}
+
+		newManifest.manifest["$"]["package"] = packageName;
+
+		const xmlBuilder = new Builder();
+		newManifestContent = xmlBuilder.buildObject(newManifest);
+
+		return newManifestContent;
+	}
+
+	private createManifestContent(packageName: string): string {
+		let newManifestContent;
+		const newManifest: any = { manifest: AndroidPluginBuildService.MANIFEST_ROOT };
+		newManifest.manifest["$"]["package"] = packageName;
+		const xmlBuilder: any = new Builder();
+		newManifestContent = xmlBuilder.buildObject(newManifest);
+
+		return newManifestContent;
+	}
+
+	private async getXml(stringContent: string): Promise<any> {
+		const promise = new Promise<any>((resolve, reject) =>
+			parseString(stringContent, (err: any, result: any) => {
+				if (err) {
+					reject(err);
+				} else {
+					resolve(result);
+				}
+			})
+		);
+
+		return promise;
+	}
+
+	private copyRecursive(source: string, destination: string): void {
+		shell.cp("-R", source, destination);
+	}
+
+	private getIncludeGradleCompileDependenciesScope(includeGradleFileContent: string): Array<string> {
+		const indexOfDependenciesScope = includeGradleFileContent.indexOf("dependencies");
+		const result: Array<string> = [];
+
+		if (indexOfDependenciesScope === -1) {
+			return result;
+		}
+
+		const indexOfRepositoriesScope = includeGradleFileContent.indexOf("repositories");
+
+		let repositoriesScope = "";
+		if (indexOfRepositoriesScope >= 0) {
+			repositoriesScope = this.getScope("repositories", includeGradleFileContent);
+			result.push(repositoriesScope);
+		}
+
+		const dependenciesScope = this.getScope("dependencies", includeGradleFileContent);
+		result.push(dependenciesScope);
+
+		return result;
+	}
+
+	private getScope(scopeName: string, content: string): string {
+		const indexOfScopeName = content.indexOf(scopeName);
+		let result = "";
+		const OPENING_BRACKET = "{";
```

Comment:
Because the indexOfScopeName can be different unline the OPENING_BRACKET symbol, but if you like I could change it ... it's not really important to me.